### ISSUE TITLE
Button component does not pass aria-current attribute to the button element

### ISSUE THREAD

**Issue by mpalmer685** [adobe/react-spectrum#6159]
### Provide a general summary of the issue here

When passing the `aria-current` prop to the Button component, it gets filtered out and does not get passed to the HTML button element.

### 🤔 Expected Behavior?

If an `aria-current` prop is passed to the Button, it should be applied to the HTML button element.

### 😯 Current Behavior

Any `aria-current` prop passed to the Button component does not make it to the HTML button element.

### 💁 Possible Solution

Update `useButton` to include the `aria-current` prop in the list of ARIA props allowed to pass through.

### 🔦 Context

We are building a Pagination component that uses `Button` instead of `Link` because the pagination may not always be tied to the current location (e.g., simple client-side pagination). We want to apply `aria-current="true"` to the button representing the current page of data, but it does not get passed correctly.

### 🖥️ Steps to Reproduce

[CodeSandbox](https://codesandbox.io/p/sandbox/react-aria-components-aria-current-l4fh2c)

### Version

"react-aria-components": "^1.1.1"

### What browsers are you seeing the problem on?

Firefox

### What operating system are you using?

macOS

**Comment by reidbarber** [comment]
This definitely seems like something we should support. We could do this in [filterDOMProps](https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/utils/src/filterDOMProps.ts), but I think it gets tricky deciding which other aria states should be available to use, and which are just implementation details of components.

**Comment by TimonVS** [comment]
Has a direction been decided yet? We are interested in this feature, and we can contribute to this. I've put up a PR with a proposed solution here: https://github.com/adobe/react-spectrum/pull/7643.
